core/iwasm/libraries/libc-wasi/sandboxed-system-primitives/src/posix.c

```diff
@@ -3270,7 +3270,11 @@ compare_address(const struct addr_pool *addr_pool_entry, const char *addr)
         return false;
     }
 
-    uint32 mask = addr_pool_entry->mask;
+    uint32 mask = 0;
+    for (int i = 0; i < addr_pool_entry->mask; i++) {
+        mask |= 1 << (31 - i);
```

Comment:
nice catch. Would u mind adding protection about `i < 32`?

Reply:
I updated the code. We have 2 options either truncating the number (`min(32, addr_pool_entry->mask)`) or rejecting invalid values. I've chosen thee latter as I thought is more natural in that case and I think it's aligned with the behavior when the ip address is invalid. Let me know if you prefer truncation, I don't mind changing that.